### PR TITLE
drop CHECK when column is dropped

### LINKED ISSUE #2391: sql: Drop CHECK constraints on table when column is dropped
The `DROP COLUMN` statement now drops `CHECK` constraints

PR: https://github.com/cockroachdb/cockroach/pull/21203

### PR DESCRIPTION
**Pull request by lhirata**
Closes #2391.

### CODE REVIEW

File: v2.0/check.md [hunk 1]
```diff
@@ -4,32 +4,33 @@ summary: The Check constraint specifies that values for the column in INSERT or
...
-- Check constraints may be specified at the column or table level and can reference other columns within the table. Internally, all column-level Check constraints are converted to table-level constraints so they can be handled consistently.
-- You can have multiple Check constraints on a single column but ideally, for performance optimization, these should be combined using the logical operators. For example
+- If you add a `CHECK` constraint to an existing table, existing values are not checked. However, any updates to those values will be.
+  {{site.data.alerts.callout_info}}In the future we plan to expand the `CHECK` constraint to include a check on any existing values in the column.{{site.data.alerts.end}}
+- `CHECK` constraints may be specified at the column or table level and can reference other columns within the table. Internally, all column-level Check constraints are converted to table-level constraints so they can be handled consistently.
+- You can have multiple `CHECK` constraints on a single column but ideally, for performance optimization, these should be combined using the logical operators. For example
```
Comment (jseldess) [inline v2.0/check.md]: nit: `For example:`

File: v2.0/check.md [hunk 2]
```diff
@@ -4,32 +4,33 @@ summary: The Check constraint specifies that values for the column in INSERT or
...
   ~~~
-  
+
   should be specified as:
-  
+
   ~~~ sql
   warranty_period INT CHECK (warranty_period BETWEEN 0 AND 24)
   ~~~
+- When a column with a `CHECK` constraint is dropped, the `CHECK` constaint is also dropped.
```
Comment (jseldess) [inline v2.0/check.md]: spelling: `constaint` > `constraint`

File: v2.0/drop-column.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ The user must have the `CREATE` [privilege](privileges.html) on the table.
 | Parameter | Description |
 |-----------|-------------|
 | `table_name` | The name of the table with the column you want to drop. |
-| `name` | The name of the column you want to drop. |
+| `name` | The name of the column you want to drop.<br><br>When a column with a `CHECK` constraint is dropped, the `CHECK` constaint is also dropped. |
```
Comment (jseldess) [inline v2.0/drop-column.md]: Same as above.

File: v2.0/check.md (hunk 1 above)
Comment (lhirata) [reply v2.0/check.md]: Done

File: v2.0/check.md (hunk 2 above)
Comment (lhirata) [reply v2.0/check.md]: Done

File: v2.0/drop-column.md (hunk 3 above)
Comment (lhirata) [reply v2.0/drop-column.md]: Done

File: v2.0/check.md [hunk 4]
```diff
@@ -4,32 +4,33 @@ summary: The Check constraint specifies that values for the column in INSERT or
...
 <div id="toc"></div>
 
 ## Details
 
-- If you add a Check constraint to an existing table, existing values are not checked. However, any updates to those values will be.
-  {{site.data.alerts.callout_info}}In the future we plan to expand the Check constraint to include a check on any existing values in the column.{{site.data.alerts.end}}
-- Check constraints may be specified at the column or table level and can reference other columns within the table. Internally, all column-level Check constraints are converted to table-level constraints so they can be handled consistently.
-- You can have multiple Check constraints on a single column but ideally, for performance optimization, these should be combined using the logical operators. For example
+- If you add a `CHECK` constraint to an existing table, existing values are not checked. However, any updates to those values will be.
+  {{site.data.alerts.callout_info}}In the future we plan to expand the `CHECK` constraint to include a check on any existing values in the column.{{site.data.alerts.end}}
```
Comment (knz) [inline v2.0/check.md]: This should link to the github issue where progress is tracked.

File: v2.0/check.md [hunk 5]
```diff
@@ -4,32 +4,33 @@ summary: The Check constraint specifies that values for the column in INSERT or
...
+- `CHECK` constraints may be specified at the column or table level and can reference other columns within the table. Internally, all column-level Check constraints are converted to table-level constraints so they can be handled consistently.
```
Comment (knz) [inline v2.0/check.md]: all column-level Check constraints -> all column-level `CHECK` constraints

File: v2.0/check.md (hunk 4 above)
Comment (lhirata) [reply v2.0/check.md]: I found [#8773](https://github.com/cockroachdb/cockroach/issues/8773) but can't find the specific issue for CHECK constraint. Do you know what issue is used to track this?

File: v2.0/check.md (hunk 5 above)
Comment (lhirata) [reply v2.0/check.md]: Done

File: v2.0/check.md (hunk 4 above)
Comment (knz) [reply v2.0/check.md]: https://github.com/cockroachdb/cockroach/issues/23663

Comment (lhirata) [reply v2.0/check.md]: 👍 thanks!

File: v2.0/check.md [hunk 6]
```diff
@@ -4,32 +4,33 @@ summary: The Check constraint specifies that values for the column in INSERT or
...
 <div id="toc"></div>
 
 ## Details
 
-- If you add a Check constraint to an existing table, existing values are not checked. However, any updates to those values will be.
-  {{site.data.alerts.callout_info}}In the future we plan to expand the Check constraint to include a check on any existing values in the column.{{site.data.alerts.end}}
-- Check constraints may be specified at the column or table level and can reference other columns within the table. Internally, all column-level Check constraints are converted to table-level constraints so they can be handled consistently.
-- You can have multiple Check constraints on a single column but ideally, for performance optimization, these should be combined using the logical operators. For example
+- If you add a `CHECK` constraint to an existing table, existing values are not checked. However, any updates to those values will be.
+  {{site.data.alerts.callout_info}}In the future we plan to [expand the `CHECK` constraint](https://github.com/cockroachdb/cockroach/issues/23663) to include a check on any existing values in the column.{{site.data.alerts.end}}
```
Comment (knz) [inline v2.0/check.md]: @dt just pointed out this point (and the callout) are plainly wrong. This limitation was lifted in cockroachdb 1.1. 
(in https://github.com/cockroachdb/cockroach/pull/9152 )
One needs to use VALIDATE to check the previous rows.

Comment (lhirata) [reply v2.0/check.md]: Ok, thanks for clarifying. I updated this bullet point for 2.0 and 1.1

**Comment by knz** [comment]
Thanks, LGTM

**Pull request merged**